pandas/tools/tests/test_pivot.py

```diff
@@ -965,6 +965,124 @@ def test_margin_ignore_dropna_bug(self):
         expected.columns = Index([3, 4, 'All'], name='b')
         tm.assert_frame_equal(actual, expected)
 
+    def test_crosstab_normalize(self):
+        # Issue 12578
+        df = pd.DataFrame({'a': [1, 2, 2, 2, 2], 'b': [3, 3, 4, 4, 4],
```

Comment:
addtl tests:

try these with all values for normalize & also do ``dropna=[True,False]`` (both values)
- all nan frame 
- empty frame